app/controllers/provider/admin/backend_apis/stats/base_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class Provider::Admin::BackendApis::Stats::BaseController < Stats::BaseController
+  include ApiAuthentication::ByAccessToken
+  self.access_token_scopes = :stats
+
+  before_action :authorize_monitoring
+  before_action :find_backend_api
+
+  activate_menu :backend_api, :monitoring
+
+  sublayout :stats
+
+  def index
+    @metrics = @backend_api.metrics.top_level
+    @methods = @backend_api.method_metrics
+
+    respond_to do |format|
+      format.html { render :index }
+      format.json { render json: metrics_with_methods }
+    end
+  end
+
+  protected
+
+  # TODO: ensure provider domain
+  def authorize_monitoring
+    authorize! :manage, :monitoring
+  end
```

Comment:
The permission to `manage` `monitoring` is defined here:
https://github.com/3scale/porta/blob/c67d5d8227a1f64c0ff2c0d91f5ea39a00cb9fb9/config/abilities/master_admin.rb#L43
https://github.com/3scale/porta/blob/c67d5d8227a1f64c0ff2c0d91f5ea39a00cb9fb9/config/abilities/provider_admin.rb#L58
https://github.com/3scale/porta/blob/c67d5d8227a1f64c0ff2c0d91f5ea39a00cb9fb9/config/abilities/provider_member.rb#L45-L46

This means that either master admin or tenant admin or tenant member with `monitoring` can access.

So to do `TODO: ensure provider domain`, there are numerous ways to do it, but one of them could be:
```ruby
def authorize_monitoring
  authorize! :manage, :monitoring
  raise CanCan::AccessDenied if current_account.master?
end
```

Reply:
This TODO has been copied from https://github.com/3scale/porta/blob/edf4cfcb170c6a6613b652796f6fab6f70674d91/app/controllers/stats/service_base_controller.rb#L18

Removed.